internal/transport/transport_test.go

```diff
@@ -2066,18 +2101,24 @@ func (s) TestClientDecodeHeaderStatusErr(t *testing.T) {
 					list: &itemList{},
 				},
 			}
-			test.metaHeaderFrame.HeadersFrame = &http2.HeadersFrame{
-				FrameHeader: http2.FrameHeader{
-					StreamID: 0,
-					Flags:    http2.FlagHeadersEndStream,
-				},
+			if test.metaHeaderFrame.HeadersFrame == nil {
+				test.metaHeaderFrame.HeadersFrame = &http2.HeadersFrame{
+					FrameHeader: http2.FrameHeader{
+						StreamID: 0,
+						Flags:    http2.FlagHeadersEndStream,
+					},
+				}
 			}
 
 			s.operateHeaders(test.metaHeaderFrame)
 
 			got := ts.status
 			want := test.wantStatus
 			if got.Code() != want.Code() || got.Message() != want.Message() {
+				t.Logf("got: %s", got.Message())
+				t.Logf("got: %d", got.Code())
+				t.Logf("want: %s", want.Message())
+				t.Logf("want: %d", want.Code())
 				t.Fatalf("operateHeaders(%v); status = %v; want %v", test.metaHeaderFrame, got, want)
```

Comment:
Why all the `Log`s?  Is `%v` not printing the code and message in a readable form?  Or can we delete these?

Reply:
`%v` was not printing the code and message in a readable form. I've adjusted the error message to print a readable message and deleted the individual logs. Let me know what you think!

Reply:
Oh let's add a `String()` method to `*status.Status` then, if you don't mind.  It should be trivial.  [This](https://github.com/grpc/grpc-go/blob/0257c8657362b76f24e7a8cfb61df48d4cb735d3/internal/status/status.go#L145) can then call it (`return e.e.String()`).

Reply:
`e.e` is an `*spb.Status` which has its own `String()` implementation. In order to call the String method I'm writing, I could change the internal error type to `*Status` (see below) with very little overall change. Is this an OK change to make?

```
type Error struct {
	e *Status
}
```

Reply:
Ah, right, it is a proto not a *Status in there.  Yes, this should be fine, unless something surprising comes up.  If you don't mind, please rename the field to `s` instead of `e`.

> Also, I'm not sure how to alert you (I know comments generate an email) so I've re-requested your review. I'm not sure if that's a good github practice or not, though!

Neither am I!  Whatever works for you is fine.  And feel free to ping via a comment if we don't respond after a couple days.

Reply:
Done!